src/sdpLoops.jl

```diff
@@ -0,0 +1,395 @@
+
+"""
+function sdp_u_w_loop(sampling_size, samples, probas, u_bounds,
+                                x_bounds, x_steps, x_dim, product_controls,
+                                dynamics, constraints, cost, Vitp, t, x, build_Ux = nothing)
+    expected_V = Inf
+    optimal_u = tuple()
+    #Loop over controls
+    if typeof(build_Ux) != Void
+        controls_search_space = build_Ux(t,x)
+    else
+        controls_search_space = product_controls
+    end
+
+    for u in controls_search_space
+
+        expected_V_u = 0.
+        count_admissible_w = 0
+
+        for w = 1:sampling_size
+            w_sample = samples[:, w]
+            proba = probas[w]
+            next_state = dynamics(t, x, u, w_sample)
+
+            if constraints(t, x, u, w_sample)&&is_next_state_feasible(next_state, x_dim, x_bounds)
+
+                count_admissible_w = count_admissible_w + proba
+
+                ind_next_state = real_index_from_variable(next_state, x_bounds,
+                                                            x_steps)
+                next_V = Vitp[ind_next_state...]
+                expected_V_u += proba*(cost(t, x, u, w_sample) + next_V)
+
+            end
+
+        end
+
+        if (count_admissible_w>0)
+
+            next_V = next_V / count_admissible_w
+
+            if (expected_V_u < expected_V)
+
+                expected_V = expected_V_u
+                optimal_u = u
+
+            end
+        end
+    end
+
+    return expected_V, optimal_u
+end
+
+"""
+Computes the optimal control at time t evaluated
+at state x at realization w in a decision hazard setting
+
+# Parameters
+* `sampling_size::int`:
+    the size of the uncertainty space
+* `samples::Array`:
+    the uncertainties realizations
+* `probas::Array`:
+    the probabilities of all the uncertainties realizations
+* `u_bounds::Array`:
+    the control variables bounds
+* `x_bounds::Array`:
+    the state variables bounds
+* `x_steps::Array`:
+    the state variables steps
+* `x_dim::int`:
+    the number of state variables
+* `product_controls::Array`:
+    the discretized control space
+* `dynamics::Function`:
+    the dynamic of the problem
+* `constraints::Function`:
+    the constraints of the problem
+* `cost::Function`:
+    the cost function of the problem
+* `V::Array`:
+    the value functions ofthe problem
+* `Vitp::Interpolations`:
+    the interpolated value function at time t+1
+* `t::float`:
+    the time step at which the value function is computed
+* `x::Array or Tuple`:
+    the state at which the value function needs to be evaluated
+* `build_Ux::Function or Void`:
+    an eventual callback to build the controls search at t and x
+
+
+# Returns
+* `optimal_u::Array`:
+    the optimal control
+
+"""
+function sdp_dh_get_u(sampling_size, samples, probas, u_bounds,
+                        x_bounds, x_steps, x_dim, product_controls,
+                        dynamics, constraints, cost, Vitp, t, x, w,
+                        build_Ux = nothing)
```

Comment:
What if we factorize the arguments of this function in a dedicated type (as `SDPLoop` or whatever)? 
I think that would DRY a bit the code, and makes the signature of this function and the following ones nicer.